### PR TITLE
Implement user activity frequency capping

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/11858

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc [hunk 1]
```diff
@@ -453,6 +463,16 @@ void AdsImpl::OnTabUpdated(
     active_tab_id_ = tab_id;
     previous_tab_url_ = active_tab_url_;
     active_tab_url_ = url;
+
+    UserActivityType user_activity_type;
+
+    if (url == "chrome://newtab/") {
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc:469]: should this be `brave://newtab`?

Comment (moritzhaller) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc:469]: also thinking out loud - this won't trigger if a user cmd + clicks on a link, which is fine I think. We could always add something like `kOpenedLinkInNewTab` if deemed useful.

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc:469]: Over time we can definitely add more event types

**Review by kylehickinson — APPROVED** [review]

**Review by moritzhaller — APPROVED** [review]
LGTM!

**Comment by btlechowski** [comment]
LGTM

Verification passed on

Brave | 1.16.46 Chromium: 86.0.4240.55 (Official Build) nightly (64-bit)
-- | --
Revision | a6d625ef6f7fe8ea0675f1cf759155a05ee1be40-refs/branch-heads/4240@{#953}
OS | Ubuntu 18.04 LTS

Verified test plan from https://github.com/brave/brave-core/pull/6703#issue-493029757


Verified ad is not shown on clean profile (0points)
Verified ad is not shown when 1 new tabs are open. (1point)
Verified ad is shown when 2 new tabs are open. (2points)
Verified ad is shown only when 1 new tab is opened and the focusing of tab is changed to another tab (2points)
Verified ad is shown after opening youtube video in a new tab (2points)
Verified ad is not shown after opening youtube video in the same tab(1point)
Verified ad is shown after browser becomes active and enters background 2times (2points)
Verified that score is reset after browser restart

When ad is not shown, logs show a generic message:
`[ads_impl.cc(876)] Ad notification not shown: Not allowed based on history`

Verified ad delivery for 2 ads per hour, 3 ads per hour, 4 ads per hour and 5 ads per hour is working as expected

**Pull request merged**